src/pandas_profiling/report/structure/overview.py

```diff
@@ -189,15 +213,48 @@ def get_dataset_column_definitions(definitions: dict) -> Container:
     )
 
 
-def get_dataset_alerts(alerts: list) -> Alerts:
+def get_dataset_alerts(config: Settings, alerts: list) -> Alerts:
+    """Obtain the alerts for the report
+
+    Args:
+        config: settings object
+        alerts: list of alerts
+
+    Returns:
+        Alerts renderable object
+    """
+    # add up alerts from multiple reports
+    if isinstance(alerts, tuple):
+        count = 0
+        combined_alerts = defaultdict(dict)
+        for i, a in enumerate(alerts):
+            for alert in a:
+                combined_alerts[f"{alert.alert_type}_{alert.column_name}"][i] = alert
```

Comment:
This seems to make the keys of `combined_alerts` keys "ragged"?

Reply:
This could indeed be solved a bit more elegantly (it is the same case with placeholders in the compare file). Refactored the code to be more clear.